packages/lesswrong/components/ultraFeed/ultraFeedSettingsValidation.ts

```diff
@@ -1,56 +1,83 @@
-    .max(1, { message: "Value must be between 0 and 1" })
-    .default(DEFAULT_SETTINGS.ultraFeedSeenPenalty),
+    .max(1, { message: "Value must be between 0 and 1" }),
   quickTakeBoost: z.number()
     .min(0.5, { message: "Value must be between 0.5 and 3.0" })
-    .max(3.0, { message: "Value must be between 0.5 and 3.0" })
-    .default(DEFAULT_SETTINGS.quickTakeBoost),
-  threadScoreAggregation: z.enum(['sum', 'max', 'logSum', 'avg'])
-    .default(DEFAULT_SETTINGS.threadScoreAggregation),
-  threadScoreFirstN: z.number().int().positive()
-    .default(DEFAULT_SETTINGS.threadScoreFirstN),
-  incognitoMode: z.boolean().default(DEFAULT_SETTINGS.incognitoMode),
-  postTruncationBreakpoints: ascendingNumbersOrNull
-    .default(DEFAULT_SETTINGS.postTruncationBreakpoints),
-  lineClampNumberOfLines: z.number().int()
-    .min(0, { message: "Value must be between 0 and 10" })
-    .max(10, { message: "Value must be between 0 and 10" })
-    .default(DEFAULT_SETTINGS.lineClampNumberOfLines),
-  commentTruncationBreakpoints: ascendingNumbersOrNull
-    .default(DEFAULT_SETTINGS.commentTruncationBreakpoints),
-  postTitlesAreModals: z.boolean().default(DEFAULT_SETTINGS.postTitlesAreModals),
+    .max(3.0, { message: "Value must be between 0.5 and 3.0" }),
+  commentSubscribedAuthorMultiplier: z.number()
+    .min(1, { message: "Value must be between 1 and 10" })
+    .max(10, { message: "Value must be between 1 and 10" }),
+  threadScoreAggregation: z.enum(['sum', 'max', 'logSum', 'avg']),
+  threadScoreFirstN: z.number().int().positive({ message: "Must be a positive integer" }),
+});
+
+const threadInterestModelSchema = z.object({
+  commentCoeff: z.number().min(0, { message: "Must be non-negative" }),
+  voteCoeff: z.number().min(0, { message: "Must be non-negative" }),
+  viewCoeff: z.number().min(0, { message: "Must be non-negative" }),
+  onReadPostFactor: z.number().min(0, { message: "Must be non-negative" }),
+  logImpactFactor: z.number(),
+  minOverallMultiplier: z.number().min(0, { message: "Must be non-negative" }),
+  maxOverallMultiplier: z.number().min(0, { message: "Must be non-negative" }),
+}).refine(data => data.minOverallMultiplier <= data.maxOverallMultiplier, {
+  message: "Min overall multiplier must be less than or equal to max overall multiplier",
+  path: ["minOverallMultiplier"], 
+});
 
-}).partial();
+const resolverSettingsSchema = z.object({
+  incognitoMode: z.boolean(),
+  sourceWeights: sourceWeightsSchema.default(DEFAULT_SETTINGS.resolverSettings.sourceWeights),
+  commentScoring: commentScoringSchema.default(DEFAULT_SETTINGS.resolverSettings.commentScoring),
+  threadInterestModel: threadInterestModelSchema.default(DEFAULT_SETTINGS.resolverSettings.threadInterestModel),
+});
+
+export const ultraFeedSettingsSchema = z.object({
+  displaySettings: displaySettingsSchema.default(DEFAULT_SETTINGS.displaySettings),
+  resolverSettings: resolverSettingsSchema,
+}).partial() as z.ZodType<UltraFeedSettingsType>; 
```

Comment:
Why does this need a type cast?  I am _very_ concerned if a zod schema declaration needs a type cast for the rest of your code to work

Reply:
Like, if there's one place you shouldn't be lying to your type system, it's in code where the entire point is that it gives you guarantees that your types line up with the runtime values.

Reply:
just shoggoth cheating...I think it works fine without